log/log.go

```diff
@@ -218,22 +252,18 @@ func output(w io.Writer, format string, a ...interface{}) {
 			}
 
 			l.loggingOff = false
-			fmt.Fprintf(w, loggingIsOff)
+			fmt.Fprintf(w, LoggingWasOff)
 		}
 
-		const waitTime = 25 * time.Millisecond
-
-		// Found out the timer can be reset down but not up. Creating
-		// the timer with one hour as the initial value.
-		l.timer.Reset(waitTime)
+		l.enqueTimer.Reset(l.stallTimeout)
 
 		// If we can't perform the write within the wait time, then
 		// let's not wait and turn off logging.
 		select {
 		case l.write <- line{w, b}:
 			atomic.AddInt32(&l.pendingWrites, 1)
-			l.timer.Stop()
-		case <-l.timer.C:
+			l.enqueTimer.Stop()
```

Comment:
The documentation shows that to use `Reset` and `Stop`, the channel needs to be drained if it hasn't fired. [Documentation](https://golang.org/pkg/time/#Timer.Reset):

```go
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```

I'm not sure if this is required here.

Reply:
This has not changed.  I just renamed `l.timer` to `l.enqueTimer`.  Which part are you referring to that is not required? I think what this does is that any time we enqueue a message into the channel we want to stop the timer because the timer is here to catch a long "insertion into the channel".  If we successfully inserted a message into the channel, the enqueTimer is no longer needed (so we stop it).  Are we on the same page or did I miss something?

Reply:
I was referring to this part in the docs
> Resetting a timer must take care not to race with the send into t.C that happens when the current timer expires. If a program has already received a value from t.C, the timer is known to have expired, and t.Reset can be used directly. If a program has not yet received a value from t.C, however, the timer must be stopped and—if Stop reports that the timer expired before being stopped—the channel explicitly drained:
```go
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```